ArduCopter/GCS_Mavlink.cpp

```diff
@@ -686,7 +694,7 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_int_do_reposition(const mavlink_co
     }
 
     // we need to do this first, as we don't want to change the flight mode unless we can also set the target
-    if (!copter.mode_guided.set_destination(request_location, false, 0, false, 0)) {
+    if (!set_destination_target(request_location, false, 0, false, 0)) {
```

Comment:
We need to be careful that this doesn't change behaviour when using guided-within-auto because the code path could be slightly different, ModeAuto::set_destination_target() vs ModeGuided::set_destination_target().